numba/targets/builtins.py

```diff
@@ -25,6 +25,14 @@ def generic_is_not(context, builder, sig, args):
     return builder.not_(is_impl(builder, args))
 
 
+@lower_builtin('is', types.Any, types.Any)
+def generic_is(context, builder, sig, args):
+    """
+    Default implementation for `x is y`
+    """
+    return cgutils.false_bit
```

Comment:
Fixed in https://github.com/numba/numba/pull/1879/commits/6b2d45781a1e3459b100b8bf47162adc47c650f9 .  The `is` impl check whether the `lhs` and `rhs` are the same type; whether they are NRT managed; whether `==` is available.